### ISSUE TITLE
[WIP] Notes for post-timestamp-PR work

### ISSUE THREAD

**Issue by losipiuk** [prestodb/presto#9450]
Notes for issues to be handled as followup work after #9385 get merged.

cc: @kokosing @findepi 

- [ ] get rid of `catch Exception` for checking if input is parsable in `DateTimeUtils`

**Comment by findepi** [comment]
Merged into https://github.com/prestodb/presto/issues/10326
